parts/linux/cloud-init/nodecustomdata.yml

```diff
@@ -95,6 +95,13 @@ write_files:
   content: !!binary |
     {{GetVariableProperty "cloudInitData" "kubeletSystemdService"}}
 
+- path: /etc/systemd/system/labels.service
```

Comment:
it seems like you've already put this in the VHDs so this bit isn't needed in this section. I suggest moving this to the {if not .IsVHDDistro}} section below.

Reply:
I noticed a few {if not .IsVHDDistro}} sections, so I added it to its own. However, should be be grouped? If so I can group it with the ones below

Reply:
My one concern with the {if not .IsVHDDistro}} is that you mentioned it could be an issue when calling the ```ensureLabels``` function if the files are not there. I'm assuming that is when it IsVHDDistro. In that case the files with be added via the other changes I made, correct? I just want to make sure the files will still be available.

Reply:
Also, for my own knowledge I expect it would be good to know what constitutes a VHDDistro, and what isn't. :)

Reply:
Yeah I am confused by what needs to be in nodecustomdata

Reply:
IsVHDDistro => AKS-managed VHDs on our production PIR/SIG (i.e aks-ubuntu-1804, aks-ubuntu-containerd-1804, etc)
Hence if you're adding the new unit files to the VHDs themselves, we don't need to add them here in nodecustomdata. But we typically fill in the gaps between here for customized OS images and Raw Ubuntu images.

Reply:
Ok, I think I am understanding this. ```{{if not .IsVHDDistro}}``` and therefore this section of ```nodecustomdata.yml``` would be added if the customer for instance used their own OS image for the node that is not managed by AKS?

One part that was concerning me was that I believe encountered something similar to the following: 
The files were not added when I vendored in these changes, and created my own standalone of RP, but didn't specify the customer VHD. However, it was still a VHD build as it was using the default image that aks manages, so this section was not used. It was simply an issue that these changes are in the default image yet.

Reply:
Talked offline. I still think there is more I could learn about IsVHDDistro, but I got a better understanding and cleared up some of my questions. Resolving